### ISSUE TITLE
Stock reservations by MRP Scheduler

### ISSUE THREAD

**Issue by ploegvde** [odoo/odoo#24441]
Impacted versions: All versions

Steps to reproduce:
1.	Today = 14-04-2018
2.	Total procurement time (delivery time vendor + security days) = 1 day
3.	Total manufacturing time Product A = 1 day
4.	BOM: Product A with Bom-Line - Component A, qty =1
5.	Create stock for Component A: Qty =10
6.	Create a manufacturing order MO1 for Product A for qty=1 for start date 01-05-2018 and validate the manufacturing order.
Odoo reserves the stock for Component A for qty=1
7.	Create a manufacturing order MO2 for Product A for qty=10 for start date 15-04-2018 and validate the manufacturing order.
Odoo reserves component A for qty=9 and says there is not enough stock for qty=1

>> Theoretically this is correct.

But when the delivery time for Component A for supplier is 1 day then there is no need to reserve the stock for MO1 now. MO2 cannot be manufactured because there is not enough stock. But that makes no sense. We can manufacture MO2 and make a purchase order for a new supply of Component A for MO1. There is enough time for that. 

So in my opinion we should make reservations as late as possible. The ultimate date to make the reservation is scheduled date of, in this example, Component A - Total procurement time Component A.
This will make it easier to deliver earlier and you do not get blocked by future MO (or DO) that does not need the stock yet.

**Comment by ploegvde** [comment]
@Yenthe666 Tanks!

**Comment by lap-odoo** [comment]
@Yenthe666 @ploegvde : Did you try to choose this setting : https://drive.google.com/a/odoo.com/file/d/1ZuJXax5vAKCJ50-rf51ySlqDnPfjh2FO/view?usp=drivesdk ? In that case, the reservation will be made based on the scheduled dates defined on the MO.

**Comment by ploegvde** [comment]
Hi, thanks for the response. Maybe better to close after response is easier :)

I don't see how this setting will help us. Can you explain your idea?

**Comment by lap-odoo** [comment]
@ploegvde Thanks to this setting, the reservation won't be made directly after the creation of the MO. When the scheduler will run, it will reserve components based on the scheduled dates of the MO. So in your case, it will reserve components for MO2 before reserving components for MO1. 

I've also created an internal feedback about that. However I don't think we will get the chance to work on that for V12.

**Comment by ploegvde** [comment]
Your solution would t work in real live. If you postpone the running of the scheduler basically all things stop. Also the creation of purchase orders and status updates of all other transfers.

**Comment by ploegvde** [comment]
Is there a way we can work on a improvement with Odoo on this?

**Comment by lap-odoo** [comment]
@ploegvde : the task is in our backlog but I don't think we will have something for v12
